### PR TITLE
[Documentation] Rework BDD guide

### PR DESCRIPTION
**Pull request by CoderMaggie**
| Q               | A
| --------------- | ---
| Doc fix?        | yes |
| New docs?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: docs/bdd/basic-usage.rst [hunk 1]
```diff
@@ -7,8 +7,12 @@ Now let us show you the flow.
 
 Describing features
 -------------------
-Let's start with writing our feature file, which will contain answers to the most important questions: Why (benefit), who (actor using the feature) and what (the feature itself). It should also include scenarios, which serve as examples of how things supposed to work.
-``features/addressing/managing_countries/adding_country.feature``
+
+Let's start with writing our feature file, which will contain answers to the most important questions:
+Why (benefit), who (actor using the feature) and what (the feature(benefit) itself).
```
Comment (GSadee) [inline docs/bdd/basic-usage.rst]: Missing space before `(`

File: docs/bdd/index.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+The BDD Guide
+=============
+
+Behaviour driven development is an approach to software development process that provides software development and management teams
+with shared tools and a shared process to collaborate on software development. The awesome part of BDD is its ubiquitous language,
+which is used to describe the software in English-like sentences of domain specific language.
+
+The application's behaviour is described by scenarios, and those scanarios are turned into automated test suites with tools such as Behat.
```
Comment (GSadee) [inline docs/bdd/index.rst]: scanarios -> sc**e**narios

File: docs/bdd/basic-usage.rst (hunk 1 above)
Comment (pamil) [reply docs/bdd/basic-usage.rst]: Shouldn't the "why" part be a reason instead of benefit?

File: docs/bdd/basic-usage.rst [hunk 3]
```diff
@@ -28,12 +32,18 @@ Scenario::
...
-A side effect of such approach is that it results in steps being really generic, therefore if we want to add another way of testing this feature for instance in the domain or api context, it will be extremely easy to apply. We just need to add a different tag (in this case "@domain") and of course implement the proper steps in the domain context of our system.
-To be more descriptive let's imagine that we want to check if a country is added properly in two ways. First we are checking if the adding works via frontend, so we are implementing steps that are clicking, opening pages, filling fields on forms and similar, but also we want to check this action regardlessly of the frontend, for that we need the domain, which allows us to perform actions only on objects.
+Pay attention to the form of these sentences. From the developer point of view they are hiding the details of the feature's implementation.
+Instead of describing "When I click on the select box And I choose France from the dropdown Then I should see the France country in the table"
```
Comment (lchrusciel) [inline docs/bdd/basic-usage.rst]: Maybe we could format these three sentences to be more consistent with the previous example?

File: docs/bdd/index.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
+The BDD Guide
+=============
+
+Behaviour driven development is an approach to software development process that provides software development and management teams
```
Comment (lchrusciel) [inline docs/bdd/index.rst]: Shouldn't we always write `Behaviour Driven Development` with the capital letter at the beginning?

File: docs/bdd/index.rst [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+with shared tools and a shared process to collaborate on software development. The awesome part of BDD is its ubiquitous language,
+which is used to describe the software in English-like sentences of domain specific language.
```
Comment (lchrusciel) [inline docs/bdd/index.rst]: Not necessary `English-like`. `Bussiness understandable`?

File: docs/bdd/index.rst [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+The application's behaviour is described by scenarios, and those scenarios are turned into automated test suites with tools such as Behat.
```
Comment (lchrusciel) [inline docs/bdd/index.rst]: The side effect of this scenarios is that, they can be turned are turned into automated test suites with tools such as Behat.

File: docs/bdd/index.rst (hunk 4 above)
Comment (CoderMaggie) [reply docs/bdd/index.rst]: In a sentence it is not needed I suppose.

File: docs/bdd/index.rst (hunk 5 above)
Comment (CoderMaggie) [reply docs/bdd/index.rst]: "domain specific language" == "understandable for business side"

File: docs/bdd/basic-usage.rst (hunk 1 above)
Comment (CoderMaggie) [reply docs/bdd/basic-usage.rst]: Benefit is the reason :D

File: docs/bdd/basic-usage.rst [hunk 7]
```diff
@@ -22,18 +26,24 @@ Scenario::
...
-A side effect of such approach is that it results in steps being really generic, therefore if we want to add another way of testing this feature for instance in the domain or api context, it will be extremely easy to apply. We just need to add a different tag (in this case "@domain") and of course implement the proper steps in the domain context of our system.
-To be more descriptive let's imagine that we want to check if a country is added properly in two ways. First we are checking if the adding works via frontend, so we are implementing steps that are clicking, opening pages, filling fields on forms and similar, but also we want to check this action regardlessly of the frontend, for that we need the domain, which allows us to perform actions only on objects.
+Pay attention to the form of these sentences. From the developer point of view they are hiding the details of the feature's implementation.
+Instead of describing "When I click on the select box And I choose United States from the dropdown Then I should see the United States country in the table"
```
Comment (pamil) [inline docs/bdd/basic-usage.rst]: I'm missing a concrete example of what should I write instead of these :)

File: docs/bdd/index.rst [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+Sylius behaviours are fully covered with Behat scenarios. There are more than 1200 scenarios in the Sylius suite, and if you want
+to understand some aspects of Sylius better, or are wondering where to click to achieve something, we strongly recommend
```
Comment (Arminek) [inline docs/bdd/index.rst]: I am not sure about this `wondering where to click to achieve something` our scenarios describe behavior of the system.

File: docs/bdd/basic-usage.rst (hunk 7 above)
Comment (CoderMaggie) [reply docs/bdd/basic-usage.rst]: @pamil the scenario is above :D

**Comment by lchrusciel** [comment]
Great work Magda! 🐳

**Pull request merged**